packages/lit-element/src/lit-element.ts

```diff
@@ -57,6 +57,28 @@ export const UpdatingElement = ReactiveElement;
 
 const DEV_MODE = true;
 
+let issueWarning: (warning: string, key?: string) => void;
+
+if (DEV_MODE) {
+  // Ensure warnings are issued only 1x, even if multiple versions of Lit
+  // are loaded.
+  const issuedWarnings: Set<string | undefined> =
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (globalThis as any)['litIssuedWarnings'] ??
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ((globalThis as any)['litIssuedWarnings'] = new Set());
+
+  // Issue a warning, de-duped by the given key.
+  issueWarning = (warning: string, key?: string) => {
+    if (!issuedWarnings.has(key)) {
+      console.warn(warning);
+      if (key) {
```

Comment:
Some of these messages are once per class or instance and I didn't want to bother managing those separate maps so went with this. Since it's DEV_MODE only it didn't seem to matter, but maybe there's a simpler way?

Reply:
Based on offline chat, we decided the extra code here isn't worth it and we'll just dedupe based on the message.